src/block/utils/FortuneDropHelper.php

```diff
@@ -0,0 +1,137 @@
+	 * @param int  $min      Minimum amount
+	 * @param int  $maxBase  Maximum amount, as if fortune level was 0
+	 *
+	 * @return int the number of items to drop
+	 */
+	public static function weighted(Item $usedItem, int $min, int $maxBase) : int{
+		if($maxBase < $min){
+			throw new \InvalidArgumentException("Maximum drop amount must be greater than or equal to minimum drop amount");
+		}
+
+		$fortuneLevel = $usedItem->getEnchantmentLevel(VanillaEnchantments::FORTUNE());
+
+		return mt_rand($min,
+			$fortuneLevel > 0 && mt_rand() / mt_getrandmax() > 2 / ($fortuneLevel + 2) ?
+				$maxBase * ($fortuneLevel + 1) :
+				$maxBase
+		);
+	}
+
+	/**
+	 * Increases the drop amount according to a binomial distribution. The function will roll maxBase+level times, and add 1
+	 * if a random number between 0-1 is less than the given probability. Each level of fortune adds one extra roll.
+	 *
+	 * As many as maxBase+level items can be dropped. This applies even if the fortune level is 0.
+	 *
+	 * @param float $chance   The chance of adding 1 to the amount for each roll, must be in the range 0-1
+	 * @param int   $min      Minimum amount
+	 * @param int   $minRolls Number of rolls if fortune level is 0, added to fortune level to calculate total rolls
+	 *
+	 * @return int the number of items to drop
+	 */
+	public static function binomial(Item $usedItem, int $min, int $minRolls = 3, float $chance = 4 / 7) : int{
+		$fortuneLevel = $usedItem->getEnchantmentLevel(VanillaEnchantments::FORTUNE());
+
+		$count = $min;
+		$rolls = $minRolls + $fortuneLevel;
+		for($i = 0; $i < $rolls; ++$i){
+			if(mt_rand() / mt_getrandmax() < $chance){
+				++$count;
+			}
+		}
+
+		return $count;
+	}
+
+	/**
+	 * Grass have a fixed chance to drop wheat seed.
+	 * Fortune level increases the maximum number of seeds that can be dropped.
+	 * A discrete uniform distribution is used to determine the number of seeds dropped.
+	 *
+	 * TODO: I'm not sure this really belongs here, but it's preferable not to duplicate this code between grass and
+	 * tall grass.
+	 *
+	 * @return Item[]
+	 */
+	public static function grass(Item $usedItem) : array{
+		if(FortuneDropHelper::bonusChance($usedItem, 8, 2)){
+			return [
+				VanillaItems::WHEAT_SEEDS()
+			];
+		}
+
+		return [];
+	}
+
+	/**
+	 * Adds the fortune level to the base max and picks a random number between the minimim and adjusted maximum.
+	 * Each amount in the range has an equal chance of being picked.
+	 *
+	 * @param int $maxBase Maximum base amount, as if the fortune level was 0
+	 *
+	 * @return int the number of items to drop
+	 */
+	public static function discrete(Item $usedItem, int $min, int $maxBase) : int{
+		if($maxBase < $min){
+			throw new \InvalidArgumentException("Minimum base drop amount must be less than or equal to maximum base drop amount");
+		}
+
+		$max = $maxBase + $usedItem->getEnchantmentLevel(VanillaEnchantments::FORTUNE());
+		return mt_rand($min, $max);
```

Comment:
that's not right. For sealantern, for example, with fortune 3, there's a 20% chance of getting 2, 3, 4, but a 40% chance of getting 5.

Reply:
This works well for those with a uniform distribution, but not for the rest who have a limit.

Reply:
I'm pretty sure it's supposed to work that way.

> Each level of Fortune increases the maximum number of prismarine crystals dropped. **The amount is capped at 5, so Fortune III simply increases the chance of getting 5 crystals.**

Reply:
Same story for melons:
> The actual amount is capped at 9, so Fortune III simply increases the chance of getting 9 slices, which is sufficient to craft the melon again from the slices.

For glowstone it's a bit more hazy, but the Java logic for these blocks is identical.

Reply:
A quick test case:
```php
$hits = [];

$runs = 10000;

$item = VanillaItems::DIAMOND_PICKAXE()->addEnchantment(new EnchantmentInstance(VanillaEnchantments::FORTUNE(), 3));
for($i = 0; $i < $runs; $i++){
	$amount = min(5, FortuneDropHelper::discrete($item, 2, 3));

	$hits[$amount] = ($hits[$amount] ?? 0) + 1;
}

ksort($hits, SORT_NUMERIC);
foreach($hits as $amount => $count){
	echo "$amount: " . (($count / $runs) * 100) . "%\n";
}
```
Output:
```
2: 20.03%
3: 19.28%
4: 20.68%
5: 40.01%
```

Reply:
> I'm pretty sure it's supposed to work that way.
> 
> > Each level of Fortune increases the maximum number of prismarine crystals dropped. **The amount is capped at 5, so Fortune III simply increases the chance of getting 5 crystals.**

that's surprising because that's not what it says on the wiki fortune'@ page.

Reply:
To paraphrase my Discord comments:

Fortune 3 on sea lantern would, if uncapped, have a 20% chance of dropping 6 crystals. However, it's capped by `min(5, amount)` in `SeaLantern`. This means that 6 gets rounded down to 5. The chance of 5 crystals is then 40%, since there are two possible results which would produce a drop count of 5.

Other blocks have the same behaviour.